src/shared/components/toast/Toast.tsx

```diff
@@ -0,0 +1,91 @@
+import classNames from 'classnames'
+import { useEffect } from 'react'
+import tw from 'twin.macro'
+
+import { Portal, usePortal } from '@/shared/components/portal'
+import CloseIcon from '@/public/svg/icons/close.svg'
+import DoubleCheckIcon from '@/public/svg/icons/double-check.svg'
+
+interface ToastProps {
+  isOpen: boolean
+  closeToast: () => void
+  autoHideDuration?: number
+  message: string
+}
+
+export const Toast = ({
+  message,
+  closeToast,
+  isOpen,
+  autoHideDuration = 3000,
+}: ToastProps) => {
+  const { container } = usePortal({
+    type: 'toast',
+    isOpen,
+    closePortal: closeToast,
+    closable: false,
+  })
+
+  // Auto hide toast
+  useEffect(() => {
+    let timeout: NodeJS.Timeout
+
+    if (isOpen) {
+      timeout = setTimeout(() => {
+        closeToast()
+      }, autoHideDuration)
```

Comment:
저는 개인적으로 consistent-return 룰을 제거하고 아래처럼 작성하는걸 더 선호합니다.
반영은 의진님이 판단해서 결정해주세요.

```suggestion
if (isOpen) {
  const timeout = setTimeout(() => {
    closeToast()
  }, autoHideDuration)
  return () => clearTimeout(timeout)
}
```

Reply:
네 이 방식이 더 깔끔하겠네요!